pkg/cmd/cli/cmd/delete.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+func NewCmdDelete(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	longDesc := `Delete a resource by filename, stdin, resource and ID, or by resources and label selector.
+
+JSON and YAML formats are accepted.
+
+If both a filename and command line arguments are passed, the command line
+arguments are used and the filename is ignored.
+
+Note that the delete command does NOT do resource version checks, so if someone
+submits an update to a resource right when you submit a delete, their update
+will be lost along with the rest of the resource.
+
+Examples:
+
+	# Delete a pod using the type and ID specified in pod.json.
+	$ %[1]s delete -f pod.json
+
+	# Delete a pod based on the type and ID in the JSON passed into stdin.
+	$ cat pod.json | %[1]s delete -f -
+
+	# Delete pods and services with label name=myLabel.
+	$ %[1]s delete pods,services -l name=myLabel
+
+	# Delete a pod with ID 1234-56-7890-234234-456456.
+	$ %[1]s delete pod 1234-56-7890-234234-456456
+
+	# Delete all pods
+	$ %[1]s delete pods --all
+`
+	var filenames util.StringList
+	cmd := &cobra.Command{
+		Use:   "delete [-f FILENAME] | [RESOURCE (ID | -l label | --all)] | [-l label]",
+		Short: "Delete a resource by filename, stdin, resource and ID, or by resources and label selector.",
+		Long:  longDesc,
+		Run: func(cmd *cobra.Command, args []string) {
+			label := cmdutil.GetFlagString(cmd, "selector")
+			if len(args) == 0 && len(label) != 0 {
+				args = []string{"bc,builds,services,imageRepositories"}
```

Comment:
> On Apr 2, 2015, at 10:07 AM, Michail Kargakis <notifications@github.com> wrote:
> 
> In pkg/cmd/cli/cmd/delete.go:
> 
> > +
> > +	# Delete a pod with ID 1234-56-7890-234234-456456.
> > +	$ %[1]s delete pod 1234-56-7890-234234-456456
> > +
> > +	# Delete all pods
> > +	$ %[1]s delete pods --all
> > +`
> > +	var filenames util.StringList
> > +	cmd := &cobra.Command{
> > +		Use:   "delete [-f FILENAME] | [RESOURCE (ID | -l label | --all)] | [-l label]",
> > +		Short: "Delete a resource by filename, stdin, resource and ID, or by resources and label selector.",
> > +		Long:  longDesc,
> > +		Run: func(cmd *cobra.Command, args []string) {
> > +			label := cmdutil.GetFlagString(cmd, "selector")
> > +			if len(args) == 0 && len(label) != 0 {
> > +				args = []string{"bc,builds,services,imageRepositories"}
> We should not hardcode this here. The proper place to handle it is in expand resource shortcut code in factory. Define one that lists all of the primary resources that an Openshift user would want and call it "all". Don't hardcode it here.
> 
> @smarterclayton I took a quick look at the shortcut expander code and it doesn't seem that I can add a shortcut for multiple resources because every resource needs to be expanded and mapped separately.
> 
> Error: no resource "buildConfigs,builds,services,imageRepositories,pods,routes,deploymentConfigs" has been defined
Then add a constant in origin/api/latest with this value and reference it.  Add a todo with it that resource expansion should happen in RESTMapper.
> —
> Reply to this email directly or view it on GitHub.
>